tests/product/test_server_upgrade.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from nose.plugins.attrib import attr
+
+from tests.product.base_product_case import BaseProductTestCase, \
+    LOCAL_RESOURCES_DIR, docker_only
+
+
+class TestServerUpgrade(BaseProductTestCase):
+
+    def setUp(self):
+        super(TestServerUpgrade, self).setUp()
+        self.setup_cluster('presto')
+
+    def start_and_assert_started(self):
+        cmd_output = self.run_prestoadmin('server start')
+        process_per_host = self.get_process_per_host(cmd_output)
+        self.assert_started(process_per_host)
+
+    def assert_upgraded(self, hosts):
+        for container in hosts:
+            # Still should have the same configs
+            self.assert_installed(container)
+            self.assert_has_default_config(container)
+            self.assert_has_default_connector(container)
+
+            # However, the new RPM removes /usr/lib/presto/lib and
```

Comment:
could you make clear that these expectations are specifically about the test RPM (not that upgrading in general deletes the plugin directory)